src/decode.rs

```diff
@@ -5161,26 +5159,8 @@ pub unsafe fn rav1d_submit_frame(c: &mut Rav1dContext) -> Rav1dResult {
     }
 
     // FIXME qsort so tiles are in order (for frame threading)
-    if f.n_tile_data_alloc < c.n_tile_data {
-        freep(&mut f.tile as *mut *mut Rav1dTileGroup as *mut c_void);
-        assert!(c.n_tile_data < i32::MAX / ::core::mem::size_of::<Rav1dTileGroup>() as c_int);
-        f.tile = malloc(c.n_tile_data as usize * ::core::mem::size_of::<Rav1dTileGroup>())
-            as *mut Rav1dTileGroup;
-        if f.tile.is_null() {
-            f.n_tile_data = 0;
-            f.n_tile_data_alloc = f.n_tile_data;
-            on_error(f, c, out_delayed);
-            return Err(ENOMEM);
-        }
-        f.n_tile_data_alloc = c.n_tile_data;
-    }
-    let num_tiles = c.n_tile_data.try_into().unwrap();
-    let f_tiles = slice::from_raw_parts_mut(f.tile, num_tiles);
-    let c_tiles = slice::from_raw_parts_mut(c.tile, num_tiles);
-    f_tiles.clone_from_slice(c_tiles);
-    c_tiles.fill_with(Default::default);
-    f.n_tile_data = c.n_tile_data;
-    c.n_tile_data = 0;
+    f.tiles.clear();
+    mem::swap(&mut f.tiles, &mut c.tiles);
```

Comment:
@fbossen, do you know why the C needed to copy from `c` to `f` rather than just swapping the two as I do here?  It seems like it would be semantically equivalent except for slight differences in the capacities, but I don't think that should matter since extra capacity is always allocated elsewhere if it's needed.